### ISSUE TITLE
`log_artifact`: move model file (optionally or by default) into Live dir

### ISSUE THREAD

**Issue by shcheklein** [iterative/dvclive#550]
In some (a lot?) of case models that are being produced are actually saved in some temporary locations, e.g. in case of YOLO it creates Git-ignored `runs/run15/weights/best.pt`. It doesn't make sense to keep model artifact there, it has to be moved somewhere. I think that we should have an option to move a model into `dvclive/models`.

I can make a PR for this. Just want to run by you @daavoo @dberenbaum @aguschin and see what you think.

**Comment by daavoo** [comment]
> In some (a lot?)

Not sure if a lot but in any case it is just people defaulting to directory-based versioning because they don't have better alternatives, not that they actually need/want it.

I am not strongly against the idea but still feel that what we should do is to instruct people to abandon "directory-based" versioning in favor of DVC experiments.

So, document that you should override the output folder to a fixed location and enable `save_dvc_exp=True`.

I think in the YOLO repo you can set a fixed output directory with `project` and `name` options

**Comment by dberenbaum** [comment]
Can we do both? We can suggest what you say @daavoo but also allow people to easily track models they already save somewhere else to make it more user-friendly/not ask them to change their code too much.

**Comment by shcheklein** [comment]
Yes, we should do both in this case I think. I don't have any opinion about changing the defaults. I see the point in the current behavior.

> I think in the YOLO repo you can set a fixed output directory with project and name options

Yep, but I'm trying to make the logger compatible with any set of the available flags. They should not be aware about `dvclive` doing something, at least they should not be changing the commands they already have.

**Comment by shcheklein** [comment]
Btw, another concern with the `log_artifact` and making `dvc add` by default is that in the YOLO repo we would be taking a lot of cache space I'm afraid - by logging the last weights every epoch. Again, not sure yet about the solution / suggestions - just sharing the information for now.

**Comment by shcheklein** [comment]
Also, how do we handle the case when model is already tracked an output? Do we simply ignore the `dvc add` error and add it do `dvc.yaml`?

**Comment by daavoo** [comment]
> Also, how do we handle the case when model is already tracked an output? Do we simply ignore the `dvc add` error and add it do `dvc.yaml`?

Yes

**Comment by aguschin** [comment]
> Can we do both? 

+1

Also, I wonder if `log_artifact` can cover cases when you save your model in the cloud bucket and want to use something like `dvc import (--no-download)`. Not sure how frequent these are though.
